Cubical/Homotopy/WhiteheadsLemma.agda

```diff
@@ -0,0 +1,447 @@
+        ( fst ((Ω^→ (2 + n)) (f , refl)))
+        ( Iso.fun (flipΩIso (1 + n))))
+    ∙ setMapFunctorial
+      ( Iso.inv (flipΩIso (1 + n)))
+      ( Iso.fun (flipΩIso (1 + n)) ∘ fst ((Ω^→ (2 + n)) (f , refl)))
+
+-- Ω (Ω→ f) ≡ Ω→ (Ω f)
+PathPΩ^→Ω : {A B : Pointed ℓ} (n : ℕ) (f : A →∙ B)
+  → PathP (λ i → (flipΩPath {A = A} n) i
+                   →∙ (flipΩPath {A = B} n) i)
+           ((Ω^→ (1 + n)) f) ((Ω^→ n) (Ω→ f))
+PathPΩ^→Ω zero f = refl
+PathPΩ^→Ω (suc n) f =
+  specialCong∙ Ω Ω→
+  ( flipΩPath n)
+  ( flipΩPath n)
+  ( Ω^→ (1 + n) f)
+  ( (Ω^→ n) (Ω→ f))
+  ( PathPΩ^→Ω n f)
+
+private
+  flipIsoSquare : {A B C D : Type ℓ} (f : A → B) (g : C → D)
+    (H : Iso A C) (J : Iso B D)
+    → Iso.inv J ∘ g ∘ (Iso.fun H) ≡ f
+    → Iso.inv (invIso J) ∘ f ∘ Iso.fun (invIso H) ≡ g
+  flipIsoSquare f g H J p =
+    sym
+    ( funExt
+      ( λ x →
+          sym ((Iso.rightInv J) (g x))
+          ∙ cong (λ y → Iso.fun J (Iso.inv J (g y)))
+                 (sym ((Iso.rightInv H) x))
+          ∙ cong (λ h → (Iso.fun J ∘ h ∘ Iso.inv H) x) p))
+
+{-
+
+Ωⁿ⁺¹ (Ω (A, a)) -------> Ωⁿ⁺¹ (Ω (B, f a))
+        |                        ^
+        |                        |
+        |                        |
+        v                        |
+Ω (Ωⁿ⁺¹ (A, a)) -------> Ω (Ωⁿ⁺¹ (B, f a))
+
+-}
+Ω+1ΩCongSquare : {A B : Type ℓ} {a : A} {n : ℕ} (f : A → B)
+  → Iso.inv (invIso (flipΩIso (1 + n)))
+   ∘ fst ((Ω^→ {A = (A , a)} (2 + n)) (f , refl))
+   ∘ Iso.fun (invIso (flipΩIso (1 + n)))
+  ≡ fst ((Ω^→ (1 + n)) (Ω→ (f , refl)))
+Ω+1ΩCongSquare {A = A} {a = a} {n = n} f =
+  flipIsoSquare
+  ( fst ((Ω^→ {A = (A , a)} (2 + n)) (f , refl)))
+  ( fst ((Ω^→ (1 + n)) (Ω→ (f , refl))))
+  ( flipΩIso (1 + n))
+  ( flipΩIso (1 + n))
+  ( ∙PathP→Square
+    ( flipΩPath (1 + n))
+    ( flipΩPath (1 + n))
+    ( (Ω^→ {A = (A , a)} (2 + n)) (f , refl))
+    ( (Ω^→ (1 + n)) (Ω→ (f , refl)))
+    ( PathPΩ^→Ω (1 + n) (f , refl)))
+
+private
+  ∙∙Lemma : {A B : Type ℓ} {f : A → B} {a b : A} (p : a ≡ b)
+            → (refl ∙∙ cong f p ∙∙ refl) ≡ cong f p
+  ∙∙Lemma {f = f} =
+    J (λ b p → (refl ∙∙ cong f p ∙∙ refl) ≡ cong f p)
+      (∙∙lCancel refl)
+
+{-
+
+πₙ (Ω (A, a)) ------> πₙ (Ω (B, f a))
+     |                        ^
+     |                        |
+     |                        |
+     v                        |
+πₙ₊₁ (A, a) ---------> πₙ₊₁ (B, f a)
+
+-}
+πHomπHomCongSquare : {A B : Type ℓ} {a : A} {n : ℕ} (f : A → B)
+  →  Iso.inv (Iso-πΩ-π (1 + n))
+   ∘ (fst (πHom {A = (A , a)} {B = (B , f a)} (1 + n) (f , refl)))
+   ∘ (Iso.fun (Iso-πΩ-π (1 + n)))
+   ≡ (fst (πHom n (cong f , refl)))
+πHomπHomCongSquare {A = A} {B = B} {a = a} {n = n} f =
+  πHomπHomCongSquareAux {A = A} {B = B} {a = a} {n = n} f
+  ∙ cong map (Ω+1ΩCongSquare {A = A} {B = B} {a = a} {n = n} f)
+  ∙ cong map (cong (fst ∘ (Ω^→ (1 + n)))
+                   ( funExt∙
+                     ( ∙∙Lemma {f = f}
+                     , JRefl ( λ b p → (refl ∙∙ cong f p ∙∙ refl) ≡ cong f p)
```

Comment:
`funExt∙ ((λ a → sym (rUnit (cong f a)))
              , rUnit (∙∙lCancel refl))`

Reply:
(no need for `JRefl`)